### PR TITLE
NN-2990: fix endpoint paging arguments

### PR DESCRIPTION
**Pull request by jamesak-oncloud9**
fix endpoint paging arguments

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceImpl.java [hunk 1]
```diff
@@ -600,8 +600,11 @@ public AdjudicationSummary getAdjudicationSummary(final Long bookingId, final Lo
     }
 
     @Override
-    public Page<VisitWithVisitors<VisitDetails>> getBookingVisitsWithVisitor(final Long bookingId, final LocalDate fromDate, final LocalDate toDate, final String visitType, Pageable pageable) {
-        return bookingService.getBookingVisitsWithVisitor(bookingId, fromDate, toDate, visitType, pageable);
+    public Page<VisitWithVisitors<VisitDetails>> getBookingVisitsWithVisitor(final Long bookingId, final LocalDate fromDate, final LocalDate toDate, final String visitType, final Integer page, final Integer size) {
```
Comment (JamieDraperUK) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceImpl.java:603]: Does this need an integration test?

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceImpl.java [hunk 2]
```diff
@@ -600,8 +600,11 @@ public AdjudicationSummary getAdjudicationSummary(final Long bookingId, final Lo
...
+        final var pageIndex = page != null ? page : 0;
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceImpl.java:604]: Not sure this is needed as annotations will set the defaults

**Review by Mjwillis — COMMENTED** [review]
Not sure null checks are needed

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceImpl.java (hunk 1 above)
Comment (petergphillips) [reply src/main/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceImpl.java:603]: normally all our endpoints should have an int test, but this doesn't look like a new one so should have one already

**Review by petergphillips — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceImpl.java (hunk 1 above)
Comment (jamesak-oncloud9) [reply src/main/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceImpl.java:603]: will check what's there already and see what I can add

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceImpl.java (hunk 2 above)
Comment (jamesak-oncloud9) [reply src/main/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceImpl.java:604]: will write a test for that and will remove if not needed, thanks

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceImpl.java (hunk 1 above)
Comment (petergphillips) [reply src/main/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceImpl.java:603]: although obviously if it doesn't then, yes please :grin:

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceImpl.java (hunk 2 above)
Comment (jamesak-oncloud9) [reply src/main/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceImpl.java]: turns out the defaults are for the benefit of swagger ui so null checks are needed, hope that's okay :)

**Review by Nimphal — APPROVED** [review]
LGTM

**Pull request merged**